docs/specs/devcontainer-reference.md

```diff
@@ -0,0 +1,144 @@
+
+A **development container** is composed of a definition (e.g. contained in a `devcontainer.json` file) that deterministically creates containers under the control of the user.
+
+## `devcontainer.json`
+
+The current metadata schema for **development containers** is contained in a `devcontainer.json` file.
+
+The `devcontainer.json` specification contains different configuration options related to the underlying orchestrator format. At the same time this specification leaves space for further development and implementation of other orchestrator mechanisms and file formats.
+
+
+# Configuration options
+
+**Development containers** currently supports multiple ways to specify the containers that create an environment. The following section describes the differences between them.
+
+## Image based
+
+Image based configurations only reference an image that should be reachable and downloadable through `docker pull` commands. Logins and tokens required for these operations are execution environment specific.. The only required parameter is `image` . The details are [here](devcontainerjson-reference.md#image-or-dockerfile-specific-properties).
+
+
+## Dockerfile based
+
+These configurations are defined as using a `Dockerfile` to define the starting point of the **development containers**. As with image based configurations, it is assumed that any base images are already reachable by **docker** when performing a `docker build` command. The only required parameter in this case is the relative reference to the `Dockerfile` in `build.dockerfile`. The details are [here](devcontainerjson-reference.md#image-or-dockerfile-specific-properties).
+
+There are multiple properties that allow users to control `docker build` works:
+
+- `build.context` 
+- `build.args`
+- `build.target`,
+- `build.cacheFrom`
+
+## Docker Compose based
+
+Docker compose configurations utilize `docker-compose`  to create and manage a set of containers required for an application. As with the other configurations, any images required for this operation are assumed to be reachable. The required parameters are:
+
+- `build.dockerComposeFile`  the reference to the docker-compose file(s) to be used.
+- `service`  this tag declares the **main** container that will be used for all other operations. Tools are assumed to also use this parameter to connect to the **development container** although they can provide facilities to connect to the other containers as required by the user.
+
+It is important to remark that due to how `docker-compose` works a lot of the properties that apply to **image** and **dockerfile** configurations can’t be set by the application and thus must be set up manually by the user using the `docker-compose` format itself.
```

Comment:
It feels like we use a combination of `docker-compose` and `Docker Compose` in this text. Could we consistently choose one? Or is there a specific pattern when we should use one vs the other?